manual/io_custom_classes/index.md

```diff
@@ -217,6 +217,8 @@ The syntax is:
 ClassDef(<ClassName>, <VersionNumber>)
 {% endhighlight %}
 
+(In the case of classes inheriting from `TObject`, it's better to use `ClassDefOverride` instead of `ClassDef`.)
```

Comment:
More than inheritance from `TObject` the choice between `ClassDef` and `ClassDefOverride` is based on whether the developer of the class has already transitioned to using the `override` keyword.  (If I remember correctly using `ClassDefOverride` for classes that does not use the override keyword will result in warning and vice et versa).

Reply:
Mmm are you sure? This is just for classes deriving from TObject derived functions (CheckConsistencyHash and so on)

Reply:
You are right it is more complicated.  It need both the current class using the keyword *and* (one of) the base class(es) having a `ClassDef` (Otherwise, indeed, it is the first declaration of those function and it needs to use the `virtual` keyword).

Reply:
```suggestion
(In the case of classes inheriting from `TObject`, or from a non-TObject base class declaring itself `ClassDef*`,
it's then better to use `ClassDefOverride` in the child instead of `ClassDef`.)
```
fyi @guitargeek